### PR TITLE
Windows: add a manifest to use UTF-8 code pages

### PR DESCRIPTION
**Pull request by wantehchang**
As of Windows Version 1903 (May 2019 Update), we can use the ActiveCodePage property in the fusion manifest for unpackaged apps to force a process to use UTF-8 as the process code page. This method requires no changes to the source code.

https://learn.microsoft.com/en-us/windows/apps/design/globalizing/use-utf8-code-page#set-a-process-code-page-to-utf-8

### CODE REVIEW

**Comment by kmilos** [comment]
Please add it via a .rc resource file instead so MinGW is supported as well? See e.g. how RStudio did it: https://github.com/rstudio/rstudio/blob/6f7af1eab02568be3de327190a96fc5dba2dc247/src/cpp/desktop/CMakeLists.txt#L308-L328

**Comment by wantehchang** [comment]
I somehow forgot that Vincent wrote https://github.com/AOMediaCodec/libavif/pull/1752 to do this. I am sorry!

Miloš: Thank you for the reminder. I didn't forget your suggestion. In the first commit of this pull request, I was aiming to make the initial solution pass the tests on all platforms. In the second commit, I tried wrapping the manifest in a resource file. It apparently works in all variants of MinGW, whether they use UCRT or MSVCRT. So the only downside of this solution is that it requires Windows 10 version 1903 instead of 1803. But I think this is a worthwhile trade-off.

**Comment by wantehchang** [comment]
I wrote this pull request originally when I looked into how to fix https://github.com/AOMediaCodec/libavif/issues/1896.

**Comment by kmilos** [comment]
Thanks Wan-Teh for working on supporting a variety of Windows toolchains.

File: apps/avifdec.c [hunk 1]
```diff
@@ -59,7 +53,7 @@ static void syntax(void)
     avifPrintVersions();
 }
 
-MAIN()
+int main(int argc, char * argv[])
```
Comment (vrabaud) [inline apps/avifdec.c:56]: I guess we use `char*argv[]` for C and `char**argv` for C++?

File: CMakeLists.txt [hunk 2]
```diff
@@ -668,20 +668,27 @@ endif()
 
 if(AVIF_BUILD_APPS)
     add_executable(avifenc apps/avifenc.c)
+    if(WIN32)
```
Comment (vrabaud) [inline CMakeLists.txt:671]: Could be factorized in a macro. Then again. `tests/` could not access it. Will refactor later.

**Comment by vrabaud** [comment]
Less code, no UTF-8 hack, LGTM !

File: apps/avifdec.c (hunk 1 above)
Comment (wantehchang) [reply apps/avifdec.c:56]: I don't know of such a difference between C and C++. I always use `char * argv[]`, but in this pull request I took care to go back to the original code, which exists for avifdec.c, avifenc.c, and are_images_equal.cc.

I don't know why I chose `char ** argv` for avifgainmaputil.cc. I may have copied it from are_images_equal.cc without thinking.

File: CMakeLists.txt (hunk 2 above)
Comment (wantehchang) [reply CMakeLists.txt:671]: Thank you. After we know this code works, we can consider using it with all executable programs, including the test programs.

File: apps/utf8.manifest [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<assembly manifestVersion="1.0" xmlns="urn:schemas-microsoft-com:asm.v1">
```
Comment (wantehchang) [inline apps/utf8.manifest:2]: I copied this file from https://learn.microsoft.com/en-us/windows/apps/design/globalizing/use-utf8-code-page#examples. The original file has this line:

```
  <assemblyIdentity type="win32" name="..." version="6.0.0.0"/>
```

I noticed that the name "..." and version "6.0.0.0" in this line look like placeholders. I think we are supposed to replace them with the name and version of avifdec, avifenc, and are_images_equal.

According to Microsoft documentation, the `assemblyIdentity` element is required in an application manifest:
https://learn.microsoft.com/en-us/windows/win32/sbscs/application-manifests

However, I deleted this line and our tests still pass. So I suggest we try going without the `assemblyIdentity` element.

**Review by wantehchang — COMMENTED** [review]
Vincent: Please review. Thanks!

When I wrote this pull request, I somehow forgot that you already wrote https://github.com/AOMediaCodec/libavif/pull/1752. I am really sorry.

File: apps/utf8.manifest (hunk 3 above)
Comment (vrabaud) [reply apps/utf8.manifest:2]: Thx for clarifying

**Review by vrabaud — APPROVED** [review]

**Pull request merged**